Extension/SafariExtensionHandler.swift

```diff
@@ -0,0 +1,98 @@
+//  ----------------------------------------------------------------
+//  Created by Yikai Zhao on 10/13/16.
+//
+//  Copyright (c) 2016-present, Yikai Zhao, Sam Deane, et al.
+//
+//  This source code is licensed under the MIT license found in the
+//  LICENSE file in the root directory of this source tree.
+//  ----------------------------------------------------------------
+
+import SafariServices
+
+
+let U2FErrorResponse = "u2f_error_response"
+
+
+let DefaultOrigin = URL(string: "https://default.origin")!
+
+/**
+ In debug only, we log out a bit of extra information to the console.
+ */
+
+func debug(_ message : String) {
+    #if DEBUG
+    NSLog(message)
```

Comment:
this is unsafe, you need to do `NSLog("%@", message)`

Reply:
It's debug only, so I'm not really worried about people cunningly crafting queries to try to exploit the extension, but point taken! 

And thanks for taking the trouble to review the code!